library/pnotes.inc

```diff
@@ -531,3 +531,14 @@ function deletePnote($id)
     sqlStatement("UPDATE pnotes SET deleted = '1', update_by = ?, update_date = NOW() WHERE id=?", array($_SESSION['authUserID'], $id));
     return true;
 }
+
+function patientNoteConvertLinks($note)
+{
+    $noteActiveLink = preg_replace('!((https://)[-a-zA-Z()0-9@:%_+.~#?&;//=]+)!i', '<a href="$1" target="_blank" rel="noopener">$1</a>', $note);
```

Comment:
Do we have anywhere in the system a standard way for link sanitation? I'm guessing we don't care about supporting unicode characters for external links right (which is a good way to cut down on a lot of spam links) but will create problems for language support.

Reply:
@adunsulag , no standard on this and yep what i am using is pretty much as simple as possible and english character centric. Figured that is safest way to proceed since this is supposed to make active links to support onc2015 and likely will be a minimally used feature. We can definitely expand on it to be more flexible down road if needed.